Aztec/Classes/Libxml2/DOM/Data/ElementNode.swift

```diff
@@ -234,6 +234,24 @@ public class ElementNode: Node {
     }
 
 
+    /// Returns the ElementNode instance, whenever there's a *single* children, of the specified nodeType (or nil otherwise).
+    ///
+    /// - Parameter type: Type of the 'single children' node to be retrieved.
+    ///
+    /// - Returns: the requested children (if it's the only children in the collection, and if the type matches), or nil otherwise.
+    ///
+    func onlyChild(ofType type: StandardElementType) -> ElementNode? {
```

Comment:
In order to aim for single-responsibility would you mind splitting this method into two steps (retrieval, filtering)?

This'd help ensure our methods are easier to compose, and also easier to read and understand.

Example:

```swift

/// This one returns first child found, if it's the only one.
///
func onlyChild() -> ElementNode? {
    guard children.count == 1 else {
        return nil
    }

    return children.first as? ElementNode
}

/// This one returns the first child found, if it's the only one and it matches the specified type.
///
func onlyChild(ofType type: StandardElementType) -> ElementNode? {
    guard let child = onlyChild(),
        child.isNodeType(type) else {
            return nil
    }

    return child
}
```